### PR TITLE
2.1/ definitions in in create new project: remove unnecessary asset dir in scaffold + update docs

### PR DESCRIPTION
**Pull request by yuhan**
### Summary & Motivation

This PR 
- changes `my-new-project/assets/` to `my-new-project/assets.py` to eliminate unnecessary files in the minimal scaffolding (per [slack convo](https://elementl-workspace.slack.com/archives/C043DECLA22/p1668195184374409?thread_ts=1668178477.056599&cid=C043DECLA22)]. 
- changes `load_assets_from_package_module` to `load_assets_from_modules` because `asset.py` is no longer a package
- updates new project doc

definitions related doc changes on this page is in the parent stack: https://github.com/dagster-io/dagster/pull/10830

### How I Tested These Changes
- locally ran `dagster project scaffold` and `dagster project scaffold-repository`; both were loaded in dagit successfully; when added an asset in `assets.py`, the new asset showed up in dagit.
- preview: https://dagster-git-yuhan-12-01-21definitionsinexamples-0f25f7-elementl.vercel.app/getting-started/create-new-project

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #10829** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10829" target="_blank">Graphite</a> 
    * **PR #10830** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10830" target="_blank">Graphite</a> 
      * **PR #10831** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10831" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10831?utm_source=stack-comment).

File: docs/content/getting-started/create-new-project.mdx [hunk 1]
```diff
@@ -189,7 +191,7 @@ dagit
 
 Open <http://localhost:3000> with your browser to see the project.
 
-Now, you can start writing assets in `my_dagster_project/assets/`, define your own ops or jobs, and include them in `my_dagster_project/__init__.py`.
+Now, you can start writing assets in `my_dagster_project/assets.pypy`, define your own ops or jobs, and include them in `my_dagster_project/__init__.py`.
```
Comment (erinkcochran87) [inline docs/content/getting-started/create-new-project.mdx:194]:
```suggestion
Now, you can start writing assets in `my_dagster_project/assets.py`, define your own ops or jobs, and include them in `my_dagster_project/__init__.py`.
```

Typo

File: docs/content/getting-started/create-new-project.mdx [hunk 2]
```diff
@@ -129,14 +129,16 @@ Inside of the `my_dagster_project/` directory, the following files and directori
...
+      <td>my_dagster_project/assets.py</td>
       <td>
-        A Python package that contains all the{" "}
-        <a href="/concepts/assets/software-defined-assets">assets</a>. It is
-        helpful to include all assets in a subpackage so that you can use{" "}
-        <PyObject object="load_assets_from_package_module" /> to load assets
-        into your project, rather than needing to add assets to the
-        project every time you define one.
+        A Python module that contains {" "}
+        <a href="/concepts/assets/software-defined-assets">Software-Defined Assets</a>.
```
Comment (erinkcochran87) [inline docs/content/getting-started/create-new-project.mdx:135]:
```suggestion
        <a href="/concepts/assets/software-defined-assets">software-defined assets</a>.
```

This isn't being used as a proper noun here, so I think we can lowercase it

File: docs/content/getting-started/create-new-project.mdx [hunk 3]
```diff
@@ -129,14 +129,16 @@ Inside of the `my_dagster_project/` directory, the following files and directori
...
+        <br />
+        <br />
+        Note: As your project grows, we recommend organizing assets in subpackages, e.g. <code>my_dagster_project/assets/analytics.py</code>, so that you can use{" "}
+        <PyObject object="load_assets_from_package_module" /> to load assets into your project and
+        group them based on the folder structure, rather than needing to manually group assets every time you define one.
```
Comment (erinkcochran87) [inline docs/content/getting-started/create-new-project.mdx:140]: > rather than needing to manually group assets every time you define one.

`group` as the verb here strikes me a bit weird. When I define a new asset, I don't think about grouping them after the fact, but more about they're included for loading. Does that make sense?

File: docs/content/getting-started/create-new-project.mdx [hunk 4]
```diff
@@ -129,14 +129,16 @@ Inside of the `my_dagster_project/` directory, the following files and directori
...
-        helpful to include all assets in a subpackage so that you can use{" "}
-        <PyObject object="load_assets_from_package_module" /> to load assets
-        into your project, rather than needing to add assets to the
-        project every time you define one.
+        A Python module that contains {" "}
+        <a href="/concepts/assets/software-defined-assets">Software-Defined Assets</a>.
+        <br />
+        <br />
+        Note: As your project grows, we recommend organizing assets in subpackages, e.g. <code>my_dagster_project/assets/analytics.py</code>, so that you can use{" "}
+        <PyObject object="load_assets_from_package_module" /> to load assets into your project and
```
Comment (erinkcochran87) [inline docs/content/getting-started/create-new-project.mdx:139]:
```
        <PyObject object="load_assets_from_package_module" /> to load assets into your project and
```

Should this be `load_assets_from_modules`?

**Review by slopp — APPROVED** [review]
LGTM! One question should we also change the treatment of ops and jobs? The project and repo.py look great, but I was a little thrown off by the sentence:

```
Now, you can start writing assets in `my_dagster_project/assets.py`, define your own ops or jobs, and include them in `my_dagster_project/__init__.py`.
```

Maybe we just kill that list bit about ops and jobs altogether?

File: docs/content/getting-started/create-new-project.mdx (hunk 4 above)
Comment (yuhan) [reply docs/content/getting-started/create-new-project.mdx:139]: unfortunately there's no `load_assets_from_modules`. the `load_assets_from_package_module` it is which is a bit mouthful but it might be too late to fix the api name :/

File: docs/content/getting-started/create-new-project.mdx (hunk 3 above)
Comment (yuhan) [reply docs/content/getting-started/create-new-project.mdx:140]: reword it to "manually add assets to the top-level definitions"

File: docs/content/getting-started/create-new-project.mdx [hunk 5]
```diff
@@ -131,14 +131,16 @@ Inside of the `my_dagster_project/` directory, the following files and directori
...
-        <a href="/concepts/assets/software-defined-assets">assets</a>. It is
-        helpful to include all assets in a subpackage so that you can use{" "}
-        <PyObject object="load_assets_from_package_module" /> to load assets
-        into your project, rather than needing to add assets to the
-        project every time you define one.
+        A Python module that contains {" "}
+        <a href="/concepts/assets/software-defined-assets">software-defined assets</a>.
+        <br />
+        <br />
+        Note: As your project grows, we recommend organizing assets in subpackages, e.g. <code>my_dagster_project/assets/analytics.py</code>, so that you can use{" "}
```
Comment (schrockn) [inline docs/content/getting-started/create-new-project.mdx:140]: `my_dagster_project/assets/analytics.py` is an example of a module not a subpackage. Subpackages are folders, so the comment is confusing.

**Review by schrockn — DISMISSED** [review]

File: docs/content/getting-started/create-new-project.mdx (hunk 5 above)
Comment (yuhan) [reply docs/content/getting-started/create-new-project.mdx]: updated to separate the wording for package and module

**Review by schrockn — APPROVED** [review]
